lib/Dialect/Torch/Transforms/ReduceOpVariants.cpp

```diff
@@ -52,17 +53,39 @@ static Type getContainerOrTensorTypeWithValueSemantics(Type type) {
   }
 }
 
+static bool
+operatorOpHasValueSemantics(OperatorOp opOp,
+                            const OwningOpRef<ModuleOp> &extraLibrary) {
+  auto opName = opOp->getAttr("name").cast<StringAttr>().getValue();
+  std::string libFuncName = (mlir::torch::Torch::getLibraryFunctionPrefix(
+                                 LibraryFunctionKind::HasValueSemantics) +
+                             Twine(opName))
+                                .str();
+  auto libFunc = extraLibrary.get().lookupSymbol<func::FuncOp>(libFuncName);
+  return bool(libFunc);
+}
+
 namespace {
 // Convert value semantic ops operating on mutable arrays to instead operate on
 // immutable tensors.
 class ConvertHasValueSemanticsOpsToValueTensors : public RewritePattern {
 public:
-  ConvertHasValueSemanticsOpsToValueTensors(MLIRContext *context)
-      : RewritePattern(MatchAnyOpTypeTag(), /*benefit=*/1, context) {}
+  ConvertHasValueSemanticsOpsToValueTensors(MLIRContext *context,
+                                            StringRef extraLibrary)
+      : RewritePattern(MatchAnyOpTypeTag(), /*benefit=*/1, context) {
+    if (!extraLibrary.empty()) {
+      loadExtraLibrary(extraLibrary.str(), context, this->extraLibrary);
```

Comment:
I'm parsing both here in the pattern and in the pass because `loadExtraLibrary` writes into a `mlir::OwningOpRef<mlir::ModuleOp>` and i'm guessing i shouldn't be passing pointers to the underlying `ModuleOp` around willy-nilly. but you guys can tell me what's the right/better way to do this.